geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegionQueue.java

```diff
@@ -464,16 +464,21 @@ private void setLatestAcknowledgedKey(Long key) {
     this.latestAcknowledgedKey.set(key);
   }
 
-  public boolean waitUntilFlushed(long timeout, TimeUnit unit) throws InterruptedException {
+  public long getLatestQueuedKey() {
+    return this.latestQueuedKey.get();
+  }
+
+  public boolean waitUntilFlushed(long latestQueuedKey, long timeout, TimeUnit unit)
+      throws InterruptedException {
     long then = System.currentTimeMillis();
     if (logger.isDebugEnabled()) {
-      logger.debug("BucketRegionQueue: waitUntilFlushed bucket=" + getId() + "; timeout=" + timeout
-          + "; unit=" + unit);
+      logger.debug("BucketRegionQueue: waitUntilFlushed bucket=" + getId() + "; latestQueuedKey="
+          + latestQueuedKey + "; timeout=" + timeout + "; unit=" + unit);
     }
     boolean result = false;
     // Wait until latestAcknowledgedKey > latestQueuedKey or the queue is empty
     if (this.initialized) {
-      long latestQueuedKeyToCheck = this.latestQueuedKey.get();
+      long latestQueuedKeyToCheck = latestQueuedKey;
```

Comment:
Not an actual problem, but we probably could just replace this variable altogether and only use latestQueueKey instead.